c_emulator/riscv_platform.h

```diff
@@ -8,6 +8,7 @@ bool sys_enable_zfinx(unit);
 bool sys_enable_writable_misa(unit);
 bool sys_enable_writable_fiom(unit);
 bool sys_enable_vext(unit);
+mach_bits sys_writable_hpm_counters(unit u);
```

Comment:
Unless I'm missing something in the spec I don't think this is needed according to privspec 3.1.10. "Hardware Performance Monitor"
> The counter registers have an arbitrary value after the hart is reset, and can be written with a given value. Any CSR write takes effect after the writing instruction has otherwise completed.

Reply:
Ah I misunderstood what this was for. I guess in the future this would be required to handle cores that have counters > N hardwired to zero, but since this is not configurable yet (riscv-config in the future?) maybe simpler to just have a hardcoded sail variable?
That variable could still be writable from the C model I guess?

Reply:
Yeah I did it like this because in our fork we actually do set this from a JSON file. Ideally we'll use riscv-config in future.

I'm not sure about poking variables in the C code from C. Feels like a bad thing to do. Especially in this repo because it still uses a copy/pasted header which can easily deviate from the actual generated code and lead to nightmarish memory corruption bugs.

Mind if I just leave it? Btw if you have a better name suggestion I'm all ears, since the one I chose is apparently confusing!